File: webviewtest/src/main/java/com/jhlee/webviewtest/TestActivity.java
package com.jhlee.webviewtest;

import android.app.Activity;
import android.content.Intent;
import android.content.res.AssetManager;
import android.net.Uri;
import android.os.Build;
import android.os.Bundle;
import android.util.Log;
import android.webkit.WebView;
import android.webkit.WebViewClient;
import android.widget.FrameLayout;
import android.window.OnBackInvokedCallback;
import android.window.OnBackInvokedDispatcher;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;

public class TestActivity extends Activity {
    private WebView webView;
    private OnBackInvokedCallback backInvokedCallback;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        Log.d("jhlee", "onCreate ");

        FrameLayout frameLayout = new FrameLayout(this);
        webView = new WebView(this);
        webView.getSettings().setJavaScriptEnabled(true);
//
        webView.setWebViewClient(new WebViewClient());



        String url = getIntent().getStringExtra("url");
        Log.d("jhlee", "TestActivity.onCreate : " + url);
//
        setContentView(frameLayout);
        frameLayout.addView(webView);
        webView.loadUrl(url);

    }
    @Override
    public void onBackPressed() {
        Log.d("jhlee", "onBackPressed ");
        finish();
    }
}

File: webviewtest/src/main/java/com/jhlee/webviewtest/TestActivity.kt
package com.jhlee.webviewtest

import android.os.Bundle
import android.util.Log
import android.webkit.ConsoleMessage
import android.webkit.WebChromeClient
import android.webkit.WebResourceRequest
import android.webkit.WebResourceResponse
import android.webkit.WebView
import android.webkit.WebViewClient
import androidx.appcompat.app.AppCompatActivity

class TestActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        Log.d("jhlee", "TestActivity_onCreate")

        setContentView(R.layout.activity_main2) // layout.xml 파일 로드

        val webView: WebView = findViewById(R.id.webview)

        webView.settings.javaScriptEnabled = true


        webView.webViewClient = object : WebViewClient() {
            override fun shouldInterceptRequest(
                view: WebView?, request: WebResourceRequest?
            ): WebResourceResponse? {
                val url = request?.url?.toString()


                Log.d("jhlee", url ?: "")
                return super.shouldInterceptRequest(view, request)
            }
        }

        webView.webChromeClient = object : WebChromeClient() {
            override fun onConsoleMessage(consoleMessage: ConsoleMessage): Boolean {
                Log.d("jhlee", "onConsoleMessage : " + consoleMessage.message().toString())
                return super.onConsoleMessage(consoleMessage)
            }
        }

//        val baseUrl = "file:///android_asset/"
        val baseUrl = ""
        val inputStream = assets.open("test2.html")
        val htmlContent = inputStream.bufferedReader().use { it.readText() }
        webView.loadDataWithBaseURL(
            baseUrl, htmlContent, "text/html", "UTF-8", null
        )
    }
}